### PR TITLE
Implement CoT detail support in TAKPacket

### PR DESCRIPTION
**Pull request by thebentern**
@niccellular once this goes live, think we should do a major version bump on the ATAK Plugin, since it's a breaking change for the non-standard CoTs?

### CODE REVIEW

File: src/modules/AtakPluginModule.cpp [hunk 1]
```diff
@@ -52,6 +52,10 @@ meshtastic_TAKPacket AtakPluginModule::cloneTAKPacketData(meshtastic_TAKPacket *
     } else if (t->which_payload_variant == meshtastic_TAKPacket_chat_tag) {
         clone.which_payload_variant = meshtastic_TAKPacket_chat_tag;
         clone.payload_variant.chat = {0};
+    } else if (t->which_payload_variant == meshtastic_TAKPacket_detail_tag) {
+        clone.which_payload_variant = meshtastic_TAKPacket_detail_tag;
+        clone.payload_variant.detail.size = t->payload_variant.detail.size;
+        memcpy(clone.payload_variant.detail.bytes, clone.payload_variant.detail.bytes, t->payload_variant.detail.size);
```
Comment (fifieldt) [inline src/modules/AtakPluginModule.cpp:58]: memcpy is source, dest, size, yeah? Is the source and dest supposed to be the same here?

Comment (thebentern) [reply src/modules/AtakPluginModule.cpp:58]: Doh! Good eyes.

File: src/modules/AtakPluginModule.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ meshtastic_TAKPacket AtakPluginModule::cloneTAKPacketData(meshtastic_TAKPacket *
     } else if (t->which_payload_variant == meshtastic_TAKPacket_detail_tag) {
         clone.which_payload_variant = meshtastic_TAKPacket_detail_tag;
         clone.payload_variant.detail.size = t->payload_variant.detail.size;
-        memcpy(clone.payload_variant.detail.bytes, clone.payload_variant.detail.bytes, t->payload_variant.detail.size);
+        memcpy(clone.payload_variant.detail.bytes, t->payload_variant.detail.size, t->payload_variant.detail.size);
```
Comment (fifieldt) [inline src/modules/AtakPluginModule.cpp:58]: maybe t->payload_variant.detail.bytes?

Comment (thebentern) [reply src/modules/AtakPluginModule.cpp]: Coffee isn't kicking in well yet this morning. I'm 0 for 2 😆

**Pull request merged**